### PR TITLE
new module vcflib/vcfallelicprimitives

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/vcflib/vcfallelicprimitives/main.nf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    conda "bioconda::vcflib=1.0.9"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vcflib:1.0.9--h146fbdb_4':
+        'biocontainers/vcflib:1.0.9--h146fbdb_4' }"
+
+    input:
+    tuple val(meta), path(vcf)
+
+    output:
+    tuple val(meta), path("*.vcf.gz")   , emit: vcf
```
Comment (chris-cheshire) [inline modules/nf-core/vcflib/vcfallelicprimitives/main.nf:15]: Whitespace

**Comment by chris-cheshire** [comment]
Looks good to me, there is some white space you could clean up around the emit statements but that's it

File: modules/nf-core/vcflib/vcfallelicprimitives/meta.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+name: "vcflib_vcfallelicprimitives"
+description: Realign reference and alternate alleles with WFA or SW, parsing out the primitive alleles into multiple VCF records. New records have IDs that reference the source record ID.  Genotypes are handled. Deletion alleles will result in haploid (missing allele) genotypes.
```
Comment (jonasscheid) [inline modules/nf-core/vcflib/vcfallelicprimitives/meta.yml:3]:
```suggestion
description: Realign reference and alternate alleles with WFA or SW, parsing out the primitive alleles into multiple VCF records. New records have IDs that reference the source record ID. Genotypes are handled. Deletion alleles will result in haploid (missing allele) genotypes.
```

**Comment by nvnieuwk** [comment]
Thanks @chris-cheshire !

**Review by jonasscheid — APPROVED** [review]
Only one nitpick. Feel free to merge afterwards

File: modules/nf-core/vcflib/vcfallelicprimitives/meta.yml (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/vcflib/vcfallelicprimitives/meta.yml:3]: Thank you!

File: modules/nf-core/vcflib/vcfallelicprimitives/main.nf [hunk 3]
```diff
@@ -0,0 +1,57 @@
+process VCFLIB_VCFALLELICPRIMITIVES {
+    tag "$meta.id"
+    label 'process_low'
+
+    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    conda "bioconda::vcflib=1.0.9"
```
Comment (SPPearce) [inline modules/nf-core/vcflib/vcfallelicprimitives/main.nf:6]: This needs to be separate environment file now.

**Review by SPPearce — CHANGES_REQUESTED** [review]
Need to move the conda environment, but the rest looks good

**Pull request closed without merging**